packages/framework-provider-azure-infrastructure/src/infrastructure/index.ts

```diff
@@ -1,39 +1,76 @@
+import { runCommand } from '@boostercloud/framework-common-helpers'
+import * as ckdtfTemplate from './templates/cdktf'
+
+export const synth = (configuration: BoosterConfig, logger: Logger): Promise<void> => synthApp(logger, configuration)
 
 export const deploy = (configuration: BoosterConfig, logger: Logger): Promise<void> => deployApp(logger, configuration)
 
 export const nuke = (configuration: BoosterConfig, logger: Logger): Promise<void> => nukeApp(logger, configuration)
 
+/**
+ * Synth the application for Azure
+ */
+async function synthApp(logger: Logger, config: BoosterConfig): Promise<void> {
+  await terraformSynth(logger, config)
+}
+
+async function uploadFile(logger: Logger, config: BoosterConfig, zipResource: ZipResource): Promise<void> {
+  logger.info('Uploading zip file')
+  const resourceGroupName = createResourceGroupName(config)
+  const functionAppName = createFunctionResourceGroupName(resourceGroupName)
+  await FunctionZip.deployZip(functionAppName, resourceGroupName, zipResource)
+  logger.info('Zip file uploaded')
+}
+
 /**
  * Deploys the application in Azure
  */
 async function deployApp(logger: Logger, config: BoosterConfig): Promise<void> {
-  const credentials = await azureCredentials()
-  const [resourceManagementClient, webSiteManagementClient] = await Promise.all([
-    createResourceManagementClient(credentials),
-    createWebSiteManagementClient(credentials),
-  ])
-  const resourceGroupName = createResourceGroupName(config)
-  await createResourceGroup(resourceGroupName, resourceManagementClient)
-  const applicationBuilder = new ApplicationStackBuilder(config)
-  await applicationBuilder.buildOn(logger, resourceManagementClient, webSiteManagementClient, resourceGroupName)
+  logger.info(`Deploying app ${config.appName}`)
+  const zipResource = await terraformSynth(logger, config)
+  const command = await runCommand(process.cwd(), 'cdktf deploy --auto-approve')
+  if (command?.stdout.includes('non-zero exit code')) {
+    return Promise.reject(`Deploy application ${config.appName} failed. Check cdktf logs`)
+  }
+  await uploadFile(logger, config, zipResource)
 }
 
 /**
  * Nuke all the resources used in the Resource Group
  */
 async function nukeApp(_logger: Logger, config: BoosterConfig): Promise<void> {
-  const credentials = await azureCredentials()
-  const resourceManagementClient = await createResourceManagementClient(credentials)
+  const command = await runCommand(process.cwd(), 'cdktf destroy --auto-approve')
+
+  if (command?.stdout.includes('non-zero exit code')) {
+    return Promise.reject(`Destroy application ${config.appName} failed. Check cdktf logs`)
+  }
+}
+
+async function generateSynthFiles(logger: Logger, config: BoosterConfig): Promise<void> {
+  logger.info('Generating cdktf files')
+  const filesToGenerate: Array<[Array<string>, string]> = [[['cdktf.json'], ckdtfTemplate.template]]
+  await Promise.all(filesToGenerate.map(renderToFile(config)))
```

Comment:
Note that `Promise.all` can fail silently, better use `Promise.allSettledAndFulfilled` (I believe there's a helper function somewhere)

Reply:
Awesome